### PR TITLE
Add an advanced activation layer for ReLU

### PR DESCRIPTION
**Pull request by tkoivisto**
The ReLU `max_value` argument can not be used currently in a layer, except a custom layer or inside a Lambda layer. Hence, similarly to LeakyReLU or for example [Softmax](https://github.com/keras-team/keras/pull/8841) this PR adds an advanced activation layer for ReLU-N, implementing a capped ReLU.

The default value for `max_value` is set to `None` meaning normal ReLU. A typical value 6.0 is included in the tests.

### CODE REVIEW

**Comment by TimZaman** [comment]
This allows for the usage of the infamous and fashioable Relu6:
https://stackoverflow.com/questions/47220595/why-the-6-in-relu6. It's so fashionable it's even its own TF layer: https://www.tensorflow.org/api_docs/python/tf/nn/relu6

**Comment by fchollet** [comment]
This is reasonable. We're already using it in a number of models. But it should be called `ReLU`.

It sounds like `LeakyReLU`, `ThresholdedReLU` and this new `ReLU` should be consolidated into a single `ReLU` layer.

**Comment by tkoivisto** [comment]
Changed to `ReLU`.

**Comment by tkoivisto** [comment]
Yeah I can take a look (later this week maybe). Did you have some specific way in mind for keeping backwards compatibility except just keeping the old layers there (I guess one could e.g. subclass from the new ReLU that will have the merged functionality)?
